src/language-js/loc.js

```diff
@@ -19,10 +35,20 @@ function locStart(node, opts) {
     }
   }
 
+  if (parser === "babel-ts" && tsKeywordTypes.has(node.type)) {
+    return node.start;
```

Comment:
```ts
const foo: (
  // comment
  never
);
```

In the above code, `start` and `end` of the `TSNeverKeywordType`  are `9` and `33` but `range` is `[11, 35]`. `range` includes parens. So in this case we should prefer `start`.

However, I'll investigate if this can be fixed in `@babel/parser` side.

Reply:
I think this is not caused by babel,we only use(and modify) `.range`, we must updated the range to wrong value.